lib/streamlit/runtime/app_session.py

```diff
@@ -667,7 +671,10 @@ def _create_file_change_message(self) -> ForwardMsg:
         return msg
 
     def _create_new_session_message(
-        self, page_script_hash: str, fragment_ids_this_run: set[str] | None = None
+        self,
+        page_script_hash: str,
+        fragment_ids_this_run: set[str] | None = None,
+        pages: dict[PageHash, PageInfo] = {},
```

Comment:
This may end up resulting in bugs down the line because the default `pages` dict is a single dict and not a new one that gets created every time a default value is needed, so if it's ever mutated the changes will persist the next time the default value is used.

It's unlikely that we'll mutate the default value here but likely don't want to write things this way anyway just to be safe.